statistics/handle/handle.go

```diff
@@ -129,25 +128,43 @@ type Handle struct {
 	sysProcTracker sessionctx.SysProcTracker
 }
 
+func (h *Handle) withRestrictedSQLExecutor(ctx context.Context, fn func(context.Context, sqlexec.RestrictedSQLExecutor) ([]chunk.Row, []*ast.ResultField, error)) ([]chunk.Row, []*ast.ResultField, error) {
+	se, err := h.pool.Get()
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+	defer h.pool.Put(se)
```

Comment:
Should we put this line before the `if` check?

Reply:
Should all internal sql executions using internal session pools ignore warnings?

Reply:
Currently we don't check warnings when executing internal sql through `(*Handle).execRestrictedSQL`, `(*Handle).execRestrictedSQLWithStatsVer` and `(*Handle).execRestrictedSQLWithSnapshot`. Maybe we can ignore warnings for now and make improvements if later warnings are needed.